ibm/resource_ibm_firewall_shared.go

```diff
@@ -0,0 +1,224 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const (
+	FwHardwarePackageType = "ADDITIONAL_SERVICES_FIREWALL"
+)
+
+func resourceIBMFirewallShared() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMFirewallSharedCreate,
+		Read:     resourceIBMFirewallSharedRead,
+		Delete:   resourceIBMFirewallSharedDelete,
+		Exists:   resourceIBMFirewallSharedExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"billing_item_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"firewall_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
I think you can two attributes instead of guest_type & guest_id
"virtual_instance_id": {
				Type:          schema.TypeInt,
				Optional:      true,
				ForceNew:      true,
				ConflictsWith: []string{"hardware_instance_id"},
			},
			"hardware_instance_id": {
				Type:          schema.TypeInt,
				Optional:      true,
				ForceNew:      true,
				ConflictsWith: []string{"virtual_instance_id"},
			},

Similarly to ibm_storage_evault resource